Feat: CP-109 Replicate dashboard charts from legacy UI to current project

Ethan225300 opened the pull request:
Completed the following changes:
Client:

Replicated dashboard charts for total profit, total assets, pending orders, and recent transactions. Ensured consistent layout, colors, and data bindings with the legacy UI. Updated ResponsiveContainer and BarChart settings to match design specifications. Aligned padding, margin, and grid classes across all dashboard cards. Validated that visual representation matches the original implementation, ensuring seamless user experience. Adjusted z-index and pointer-events settings to resolve click issues on chart interactions. 
Resolve CP-109
<img width="1280" alt="2fa0d2ae727d31529c56d3a0a1b0920" src="https://github.com/user-attachments/assets/7650dfa1-8efd-48de-86e1-33e04992bffb">
<img width="1280" alt="9109dcda34fcf94af780161c4d5471a" src="https://github.com/user-attachments/assets/0063a71e-6264-4144-9bd3-685417fed25e">

<!-- link to tickets -->

---

## What happens here?

## How do I know this is working?

## Any notes?

<!--
  New PR Checklist

- Have you tag the right reviewer?
- Have you dm the reviewer?
- Maybe @ the reviewer in channel just to be safe

- is CI passing in your pr?
- Have you update the ticket with the right status?
-->

NorrisWu0 commented on apps/client/src/app/[locale]/(protected)/dashboard/components/PendingOrders.tsx:
```diff
@@ -0,0 +1,46 @@
+const PendingOrders = () => (
+  <div className="col-span-1 xl:col-span-1 lg:col-span-2 md:col-span-3"> 
+    <div className="bg-white pt-[25px] pb-[38px] px-[25px] rounded-lg md:h-[125px]"
+    style={{ boxShadow: '0 10px 30px 1px rgba(0, 0, 0, 0.06)' }}>
```
this can be done with tailwind arbitrary value yeah? best not to avoid inline style if possible

NorrisWu0 commented on apps/client/src/app/[locale]/(protected)/dashboard/components/PendingOrders.tsx:
```diff
@@ -0,0 +1,46 @@
+        >
+          25
+        </h2>
+        <TrendingUpIcon className="text-gray-300 w-6 h-6" />
```
`react-icons` has mdi too, use that lib instead

Btw. Check common module for a `<Icon/>` component

NorrisWu0 commented on apps/client/src/app/[locale]/(protected)/dashboard/components/PendingOrders.tsx:
```diff
@@ -0,0 +1,46 @@
+        <TrendingUpIcon className="text-gray-300 w-6 h-6" />
+      </div>
+      <h5
+        className="text-[12px] font-medium text-left opacity-70 mt-[-4px] uppercase text-primary-600 font-sans"
```
Check what we have in tw-config when using font.

Ethan225300 replied:
> Check what we have in tw-config when using font.

NorrisWu0 commented:
Yeah. Maybe change how you style the component first, that's quite some changes to get going already. 😉

Maybe ask @s3713434 or @SeijaHuang or @Maaayhan for help if needed.

NorrisWu0 commented on apps/client/package.json:
```diff
@@ -20,14 +20,16 @@
     "clsx": "^2.1.1",
     "graphql": "^16.9.0",
     "graphql-request": "^7.1.0",
+    "mdi-react": "^9.4.0",
```
remember to remove package if we not using them anymore

NorrisWu0 replied:
remember to remove this if we don't use them anymore

NorrisWu0 commented on apps/client/src/app/[locale]/(protected)/dashboard/components/PendingOrders.tsx:
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { MdTrendingUp } from 'react-icons/md';
+
+const PendingOrders = () => (
```
okay, so all 4 cards/charts looks fairly similar in style and functionality yeah? maybe try to make them a component with different variants? 

then on top of that you could also consider making them "stateless" by taking in data props. An idea would look something like this

```tsx
type Props = {
  title: string,
  subtitle?: string,
  theme: "red" | "blue" | "green" // or how ever want to call this?
} & (
  {
    type: "percentage"
    percentage: number
  } | {
    type: "bar-chart"
    someData: ????
  }
)
```


let me know if you need more clarification on this

Commit:
Feat: Replicate dashboard charts from legacy UI to current project
Completed the following changes:
Client:

Replicated dashboard charts for total profit, total assets, pending orders, and recent transactions.
Ensured consistent layout, colors, and data bindings with the legacy UI.
Updated ResponsiveContainer and BarChart settings to match design specifications.
Aligned padding, margin, and grid classes across all dashboard cards.
Validated that visual representation matches the original implementation, ensuring seamless user experience.
Adjusted z-index and pointer-events settings to resolve click issues on chart interactions.
Resolve CP-109

Chris-Wang reviewed (approved):
Thanks for the review @NorrisWu0 
Good job @Ethan225300 
LGTM
